frontend-react/src/AppRouter.tsx

```diff
@@ -41,6 +43,52 @@ export enum FeatureName {
 }
 
 export const AppRouter = () => {
+    const {
+        sessionStartTime,
+        setSessionStartTime,
+        sessionTimeAggregate,
+        setSessionTimeAggregate,
+    } = useSessionContext();
+
+    useEffect(() => {
+        const onUnload = () => {
+            trackAppInsightEvent(EventName.SESSION, {
+                sessionLength:
+                    (new Date().getTime() - sessionStartTime.getTime()) / 1000 +
```

Comment:
Can we create a function to calculate the sessionLength since its used in multiple places?